scripts/linux/publishReleasePackages.sh

```diff
@@ -76,36 +90,38 @@ process_args() {
         elif [ $save_next_arg -eq 4 ]; then
             SERVER="$arg"
             save_next_arg=0
+        elif [ $save_next_arg -eq 5 ]; then
+            GITHUB_PAT="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 6 ]; then
+            VERSION="$arg"
+            save_next_arg=0   
+        elif [ $save_next_arg -eq 7 ]; then
+            SKIP_UPLOAD="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 8 ]; then
+            BRANCH_NAME="$arg"
+            save_next_arg=0   
         else
             case "$arg" in
             "-h" | "--help") usage ;;
             "-p" | "--packageos") save_next_arg=1 ;;
             "-d" | "--dir") save_next_arg=2 ;;
             "-w" | "--wdir") save_next_arg=3 ;;
             "-s" | "--server") save_next_arg=4 ;;
+            "-g" | "--ghubpat") save_next_arg=5 ;;
+            "-v" | "--version") save_next_arg=6 ;;
+            "-u" | "--skip-upload") save_next_arg=7 ;;
+            "-b" | "--branch-name") save_next_arg=8 ;;
             *) usage ;;
             esac
         fi
     done
 }
 
-###############################################################################
-# Process Start
-###############################################################################
-
-process_args "$@"
-check_os
-check_dir
-check_server
-echo "OS is $OS_NAME"
-echo "Version is $OS_VERSION"
-echo "Work Dir is $WDIR"
-echo "Package OS DIR is $DIR"
-
-#Debug View of Package Dir Path
-ls -al $DIR 
-
-#Cleanup
+publish_to_microsoft_repo()
+{
+    #Cleanup
 sudo rm -rf $WDIR/private-key.pem || true
 sudo rm -rf $WDIR/$OS_NAME-$OS_VERSION-multi-aad.json || true
```

Comment:
For line 142, could you make it display the `location` ie "/v2/packages/queue/<someHash>" out to the log as well please?  This will be crucial if we need to revoke the package.

For upload check loop line 156-175, great idea, however it may not be sufficient. There has been a case where the server claimed to have packages upload, but it only upload to ONE server and was not sync across other server for packages.microsoft.com.  We caught the issue by going to different targeted URI i.e. `https://packages.microsoft.com/ubuntu/18.04/multiarch/prod/pool/main/i/iotedge/ ` and download the latest, and run `dpkg --info` to verify the version is correct (same for rpm `rpm -qi` ?).

Reply:
I dont think the repoclient tool we have has the option of displaying package queue. What do you mean by location? Is it the path in packages.microsoft.com where packages are stored?

I can definitely add further optimization to the upload in terms of validation, But I believe it would be outside the scope of this PR. I'll address that in a separate PR

Reply:
I did a follow up and verified that the response I get from 'request check' is similar to what i get from 'packages check'(Which calls v3/packages/queue). 

I like the idea that we can verify via dpkg , but I'l leave it as future optimization :)

Reply:
Sounds good. For now, I'll just have to do the post-release "smoke" test manually. That's fine with me for the 1st iteration.